main/jlp/widgets/widget_factory.cpp

```
@@ -2480,6 +2497,294 @@ lv_obj_t* build_voice(BuildCtx& ctx, JsonObjectConst spec, std::string* err) {
+  } else {
+    lv_label_set_text(ph, (c->started && s.connected) ? "no signal" : "connecting…");
+    lv_obj_clear_flag(ph, LV_OBJ_FLAG_HIDDEN);
+  }
+}
+
+void stream_set_visible(StreamCtx* c, bool vis) {
+  if (vis == c->visible) return;
+  c->visible = vis;
+  if (vis) {
+    stream_try_start(c);
+    if (!c->watch_timer) {
```

Clang (14.0.6)

[warning] implicit conversion 'uint32_t' (aka 'unsigned int') -> bool

(readability-implicit-bool-conversion)